### PR TITLE
PSP: godoc fixes

### PR DESCRIPTION
**Pull request by php-coder**
**What this PR does / why we need it**:
This PR fixes typos and wrong information along with updating comments to follow godoc convention.

### CODE REVIEW

**Comment by thockin** [comment]
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes/60792)

**Comment by php-coder** [comment]
/kind cleanup
/release-note-none

PTAL @kubernetes/api-reviewers 

CC @simo5

**Comment by php-coder** [comment]
Verification fails with a strange error that could be related to infra issues rather than typo fixes in the comments.

```
I0305 14:18:33.655126    5516 parse.go:418] type checker: /usr/local/go/src/crypto/cipher/xor.go:12:36: cannot use uintptr(0) (constant 0 of type uintptr) as unsafe.ArbitraryType value in argument to unsafe.Sizeof
I0305 14:18:33.655174    5516 parse.go:418] type checker: /usr/local/go/src/crypto/cipher/xor.go:12:18: int(unsafe.Sizeof(uintptr(0))) (value of type int) is not constant
I0305 14:18:33.655266    5516 parse.go:367] type checking encountered some issues in "crypto/cipher", but ignoring.
I0305 14:18:33.700725    5516 parse.go:418] type checker: /usr/local/go/src/crypto/tls/cipher_suites.go:18:2: could not import golang_org/x/crypto/chacha20poly1305 (unable to import "golang_org/x/crypto/chacha20poly1305": cannot find package "golang_org/x/crypto/chacha20poly1305" in any of:
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/k8s.io/kubernetes/vendor/golang_org/x/crypto/chacha20poly1305 (vendor tree)
	/usr/local/go/src/golang_org/x/crypto/chacha20poly1305 (from $GOROOT)
	/go/src/k8s.io/kubernetes/_output/dockerized/go/src/golang_org/x/crypto/chacha20poly1305 (from $GOPATH))
```

@sttts Any chance that you know what's happening here? Or who I should ping in this case?

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 1]
```diff
@@ -1001,7 +1001,7 @@ type HostPortRange struct {
 	Max int32 `json:"max" protobuf:"varint,2,opt,name=max"`
 }
 
-// SELinux  Strategy Options defines the strategy type and any options used to create the strategy.
+// SELinux Strategy Options defines the strategy type and any options used to create the strategy.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go]: if you're going to change this, make them actually match godoc conventions and start with the type name

Comment (php-coder) [reply staging/src/k8s.io/api/extensions/v1beta1/types.go]: Shall I also update other PSP-related fields to follow this conventions?

**Comment by php-coder** [comment]
/hold

I'll revise it later.

**Comment by php-coder** [comment]
For unknown reason I had to run `hack/update-generated-protobuf.sh`. I didn't change anything except comments so I was surprised that protobuf structure has been changed.

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 2]
```diff
@@ -896,25 +896,25 @@ type PodSecurityPolicySpec struct {
 	// You must not list a capability in both AllowedCapabilities and RequiredDropCapabilities.
 	// +optional
 	AllowedCapabilities []v1.Capability `json:"allowedCapabilities,omitempty" protobuf:"bytes,4,rep,name=allowedCapabilities,casttype=k8s.io/api/core/v1.Capability"`
-	// volumes is a white list of allowed volume plugins.  Empty indicates that all plugins
-	// may be used.
+	// Volumes is a white list of allowed volume plugins. Empty indicates that
+	// none volumes may be used. To allow all volumes you may use '*'.
```
Comment (tallclair) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go]: s/none/no/

File: staging/src/k8s.io/api/policy/v1beta1/types.go [hunk 3]
```diff
@@ -153,25 +153,25 @@ type PodSecurityPolicySpec struct {
 	// You must not list a capability in both AllowedCapabilities and RequiredDropCapabilities.
 	// +optional
 	AllowedCapabilities []v1.Capability `json:"allowedCapabilities,omitempty" protobuf:"bytes,4,rep,name=allowedCapabilities,casttype=k8s.io/api/core/v1.Capability"`
-	// volumes is a white list of allowed volume plugins.  Empty indicates that all plugins
-	// may be used.
+	// Volumes is a white list of allowed volume plugins. Empty indicates that
+	// none volumes may be used. To allow all volumes you may use '*'.
```
Comment (tallclair) [inline staging/src/k8s.io/api/policy/v1beta1/types.go]: s/none/no/

**Comment by php-coder** [comment]
@tallclair Fixed, thanks! I've also updated `pkg/apis/extensions/types.go` file. PTAL.

**Comment by php-coder** [comment]
@thockin PTAL

**Comment by php-coder** [comment]
@liggitt Would you like to review it once more? It's green and needs approval.

File: staging/src/k8s.io/api/extensions/v1beta1/types.go [hunk 4]
```diff
@@ -876,9 +876,9 @@ type PodSecurityPolicy struct {
 	Spec PodSecurityPolicySpec `json:"spec,omitempty" protobuf:"bytes,2,opt,name=spec"`
 }
 
-// Pod Security Policy Spec defines the policy enforced.
+// PodSecurityPolicySpec defines the policy enforced.
 type PodSecurityPolicySpec struct {
-	// privileged determines if a pod can request to be run as privileged.
+	// Privileged determines if a pod can request to be run as privileged.
```
Comment (liggitt) [inline staging/src/k8s.io/api/extensions/v1beta1/types.go]: leave these lowercase... we want api documentation to match the json fields

Comment (php-coder) [reply staging/src/k8s.io/api/extensions/v1beta1/types.go]: Thanks! I reverted these changes and modified other places to use lowercase. PTAL.

**Pull request merged**